src/components/OptionsList/BaseOptionsList.js

```diff
@@ -35,276 +35,277 @@ const defaultProps = {
     ...optionsListDefaultProps,
 };
 
-function BaseOptionsList({
-    keyboardDismissMode,
-    onScrollBeginDrag,
-    onScroll,
-    listStyles,
-    focusedIndex,
-    selectedOptions,
-    headerMessage,
-    isLoading,
-    sections,
-    onLayout,
-    hideSectionHeaders,
-    shouldHaveOptionSeparator,
-    showTitleTooltip,
-    optionHoveredStyle,
-    contentContainerStyles,
-    sectionHeaderStyle,
-    showScrollIndicator,
-    listContainerStyles,
-    shouldDisableRowInnerPadding,
-    shouldPreventDefaultFocusOnSelectRow,
-    disableFocusOptions,
-    canSelectMultipleOptions,
-    shouldShowMultipleOptionSelectorAsButton,
-    multipleOptionSelectorButtonText,
-    onAddToSelection,
-    highlightSelectedOptions,
-    onSelectRow,
-    boldStyle,
-    isDisabled,
-    innerRef,
-    isRowMultilineSupported,
-    isLoadingNewOptions,
-    nestedScrollEnabled,
-    bounces,
-}) {
-    const flattenedData = useRef();
-    const previousSections = usePrevious(sections);
-    const didLayout = useRef(false);
-
-    /**
-     * This helper function is used to memoize the computation needed for getItemLayout. It is run whenever section data changes.
-     *
-     * @returns {Array<Object>}
-     */
-    const buildFlatSectionArray = () => {
-        let offset = 0;
-
-        // Start with just an empty list header
-        const flatArray = [{length: 0, offset}];
-
-        // Build the flat array
-        for (let sectionIndex = 0; sectionIndex < sections.length; sectionIndex++) {
-            const section = sections[sectionIndex];
-
-            // Add the section header
-            const sectionHeaderHeight = section.title && !hideSectionHeaders ? variables.optionsListSectionHeaderHeight : 0;
-            flatArray.push({length: sectionHeaderHeight, offset});
-            offset += sectionHeaderHeight;
-
-            // Add section items
-            for (let i = 0; i < section.data.length; i++) {
-                let fullOptionHeight = variables.optionRowHeight;
-                if (i > 0 && shouldHaveOptionSeparator) {
-                    fullOptionHeight += variables.borderTopWidth;
+const BaseOptionsList = forwardRef(
```

Comment:
The main change here is refactoring the `forwardRef`. The second change is memoizing the `listContentContainerStyle `.